### ISSUE TITLE
importing from xmipp3.base fails while performing template matching

### ISSUE THREAD

**Issue by noushinha** [scipion-em/scipion-em-emantomo#127]
I am trying to use the template matching. I load a tomogram of size 500x500x250 and 10A using tomo, and a volume of the same voxel size using pwem. Applying the template matching to these input data produces the following output log:

```
00001:   Logging configured. STDOUT --> /home/visual//ScipionUserData/projects/sta/Runs/000513_EmanProtTemplateMatching/logs/run.stdout , STDERR --> /home/visual//ScipionUserData/projects/sta/Runs/000513_EmanProtTemplateMatching/logs/run.stderr
00002:   RUNNING PROTOCOL -----------------
00003:   Protocol starts
00004:   Hostname: korlan.zib.de
00005:   PID: 2421873
00006:   pyworkflow: 3.9.2
00007:   plugin: emantomo - https://github.com/scipion-em/scipion-em-eman2
00008:   plugin v: 3.3.5(production)
00009:   plugin binary v: 2.99.55
00010:   currentDir: /home/visual/ScipionUserData/projects/sta
00011:   workingDir: Runs/000513_EmanProtTemplateMatching
00012:   runMode: Restart
00013:             MPI: 1
00014:         threads: 4
00015:    Starting at step: 1
00016:    Running steps 
00017:   STARTED: prepareEmanPrj, step 1, time 2024-10-14 13:49:06.596634
00018:   ** Running command: **
00019:    eval "$(/scratch/htc/miniconda3/condabin/conda shell.bash hook)"&& conda activate eman-2.99.55 &&  e2proc3d.py  /home/visual/ScipionUserData/projects/sta/Runs/000002_ProtImportTomograms/extra/TS_0001.mrc tomograms/TS_0001.hdf --apix 13.480
00020:   FAILED: prepareEmanPrj, step 1, time 2024-10-14 13:49:08.440093
00021:   *** Last status is failed 
00022:   ------------------- PROTOCOL FAILED (DONE 1/4)
```

Also in the project log, I see the following:

`error when importing from xmipp3.protocols: cannot import name 'isMdEmpty' from 'xmipp3.base'`

Would you please tell me what I am missing here?

![Screenshot from 2024-10-14 14-00-36](https://github.com/user-attachments/assets/146d44f6-455c-4e3f-90ac-2b2a5f175e58)

**Comment by JorMaister** [comment]
Hi, @noushinha, 

Could you paste the output log > run.stderr? It can be found in the lower panel of the main project. On the other hand, maybe you don't know the Discord server we have. Developers and users normally join in and the community is very active. We also have different specific channels per plugin, etc. It's the fastest way to get feedback, but github issues are more than welcome. It was just to let you know the possible ways to reach us. In case you are interested to join our server, here's an invitation link:

https://discord.gg/qD5D3zmN

**Comment by noushinha** [comment]
Thanks a lot for the reply and invitation.

Here is the output from run.stderr:

```
00001:   /bin/sh: 3: /scratch/htc/miniconda3/envs/scipion3/etc/conda/deactivate.d/openjdk_deactivate.sh: [[: not found
00002:   /bin/sh: 9: /scratch/htc/miniconda3/envs/scipion3/etc/conda/deactivate.d/openjdk_deactivate.sh: [[: not found
00003:   
00004:   EnvironmentNameNotFound: Could not find conda environment: eman-2.99.55
00005:   You can list all discoverable environments with `conda info --envs`.
00006:   
00007:   
00008:   Traceback (most recent call last):
00009:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/protocol/protocol.py", line 207, in run
00010:       self._run()
00011:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/protocol/protocol.py", line 257, in _run
00012:       resultFiles = self._runFunc()
00013:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/protocol/protocol.py", line 253, in _runFunc
00014:       return self._func(*self._args)
00015:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/site-packages/emantomo/protocols/protocol_template_matching.py", line 136, in prepareEmanPrj
00016:       self.convertOrLink(inTomoName, tomo.getTsId(), TOMOGRAMS_DIR, sRate)
00017:     File "/scratch/htc//miniconda3/envs/scipion3/lib/python3.8/site-packages/emantomo/protocols/protocol_base.py", line 204, in convertOrLink
00018:       self.convertBetweenHdfAndMrc(abspath(inFile), outFile, extraArgs=f'--apix {sRate:.3f}')
00019:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/site-packages/emantomo/protocols/protocol_base.py", line 210, in convertBetweenHdfAndMrc
00020:       self.runJob(program, args + extraArgs, cwd=self._getExtraPath())
00021:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/protocol/protocol.py", line 1534, in runJob
00022:       self._stepsExecutor.runJob(self._log, program, arguments, **kwargs)
00023:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/protocol/executor.py", line 68, in runJob
00024:       process.runJob(log, programName, params,
00025:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/utils/process.py", line 56, in runJob
00026:       return runCommand(command, env=env, cwd=cwd, executable=executable)
00027:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/site-packages/pyworkflow/utils/process.py", line 71, in runCommand
00028:       check_call(command, shell=True, stdout=sys.stdout, stderr=sys.stderr,
00029:     File "/scratch/htc/miniconda3/envs/scipion3/lib/python3.8/subprocess.py", line 364, in check_call
00030:       raise CalledProcessError(retcode, cmd)
00031:   subprocess.CalledProcessError: Command ' eval "$(/scratch/htc/miniconda3/condabin/conda shell.bash hook)"&& conda activate eman-2.99.55 &&  e2proc3d.py  /home/visual/ScipionUserData/projects/sta/Runs/000839_ProtImportTomograms/extra/TS_0001.mrc tomograms/TS_0001.hdf --apix 13.480' returned non-zero exit status 1.
00032:   Protocol failed: Command ' eval "$(/scratch/htc/miniconda3/condabin/conda shell.bash hook)"&& conda activate eman-2.99.55 &&  e2proc3d.py  /home/visual/ScipionUserData/projects/sta/Runs/000839_ProtImportTomograms/extra/TS_0001.mrc tomograms/TS_0001.hdf --apix 13.480' returned non-zero exit status 1.
```

**Comment by noushinha** [comment]
solved by "`scipion3 installb eman-2.99.55 -j 5`"
